### DISCUSSION TITLE
Energy meters in 3-Phase configurations show wrong amounts

### DISCUSSION THREAD

**Discussion by Tho-Mat** [jasonacox/Powerwall-Dashboard#526] [General]
My home is connected with 3-phases to the grid. 
I have a saldoing grid-meter, that means if i export energy on some phases and import energy on other phases
the meter only counts the difference. 
So, if Phase1-Energy + Phase2-Energy + Phase3-Energy is >0 at a time, the import meter rises.
Or, if Phase1-Energy + Phase2-Energy + Phase3-Energy is <0 at a time, the export meter rises.

I always was wondering, why the powerwall values METER_X_LifetimeEnergyImport and METER_X_LifetimeEnergyExport do not match my home grid-meter. 
On the other hand the values shown in the Tesla app do, with only a small error.

Example: 2024-09-14 
grid-meter of the grid-provider: 
        import: 0,41195kWh export: 20,5937kWh
App: 
        import 0,3kWh export 20,5kWh
Meter_X: 
        import 4,606kWh export: 24,817kWh

So what does Tesla do in the app?

I recorded the meter values every 5min and then calculated the difference of import - export. 
If this is negative, then I assume an export of the kWh difference. 
If this is positive, then I assume an import of the kWh difference.
I now made a summ over all positive values of the day and get 0,293kWh for import.
Also the summ over all negative values gives me 20,504kWh for export.
I do not know if Tesla uses a 5 min interval, but using this gives a good approximation. 

P.S.: 
This may also apply to (3-phase) Y_Meter or PW3. 
The actual METER_X_LifetimeEnergyImport is not senseless. 
In the example above you can see, that in an ideal island configuration 
I would at least have consumed round about 4kWh more energy.
Unfortunately the METER_X_LifetimeEnergy does not show the total amount of imported/exported energy.
When i look at the three METER_X_CT?_InstRealPower values 
the total amount of imported energy is 10,7kWh and 30,9kWh for exported energy.

**Comment by JanSchr** [comment]
Same issues here, took me some time to realize L3 is missing (I have two powerwalls 3 on L1 and L2) and so home usage is missing L3 data (it's always to low). 

I guess there's no other way than installing a third PW3 on L3 to fix that? Or did you found some kind of fix/workaround?

**Reply by jasonacox** [reply]
Fascinating. I confess, I've struggled a lot to try to figure out how to support all the different 3-phase install variations. Clearly, Tesla is managing the right aggregation on their side and potentially exposing the key metrics to inform the right equations to use. Unfortunately, my setup is super simple so I'm at a loss for that discovery and testing. If anyone figures out a better way to accommodate the 3-phase variates, please let me know (or better yet, submit PR at pypowerwall or here :).
